packages/frontend/src/features/sqlRunner/hooks/useSqlQueryRun.tsx

```diff
@@ -0,0 +1,168 @@
+import { useMemo } from 'react';
+import { lightdashApi } from '../../../api';
+import useToaster from '../../../hooks/toaster/useToaster';
+import { getSchedulerJobStatus } from '../../scheduler/hooks/useScheduler';
+import { useAppSelector } from '../store/hooks';
+
+const scheduleSqlJob = async ({
+    projectUuid,
+    sql,
+}: {
+    projectUuid: string;
+    sql: SqlRunnerBody['sql'];
+}) =>
+    lightdashApi<ApiJobScheduledResponse['results']>({
+        url: `/projects/${projectUuid}/sqlRunner/run`,
+        method: 'POST',
+        body: JSON.stringify({ sql }),
+    });
+
+/**
+ * Gets the SQL query results from the server
+ *
+ * Steps:
+ * 1. Schedule the SQL query job
+ * 2. Get the status of the scheduled job
+ * 3. Fetch the results of the job
+ */
+export const useSqlQueryRun = () => {
+    const { showToastError } = useToaster();
+    const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
+    const queryClient = useQueryClient();
+    const sqlQueryRunMutation = useMutation<
+        ApiJobScheduledResponse['results'],
+        ApiError,
+        {
+            projectUuid: string;
+            sql: SqlRunnerBody['sql'];
+        }
+    >(({ sql }) => scheduleSqlJob({ projectUuid, sql }), {
+        mutationKey: ['sqlRunner', 'run'],
+    });
+
+    const { data: sqlQueryJob } = sqlQueryRunMutation;
+
+    const { data: scheduledDeliveryJobStatus } = useQuery<
+        ApiJobStatusResponse['results'] | undefined,
+        ApiError
+    >(
+        ['jobStatus', sqlQueryJob?.jobId],
+        () => {
+            if (!sqlQueryJob?.jobId) return;
+            return getSchedulerJobStatus(sqlQueryJob.jobId);
+        },
+        {
+            refetchInterval: (data) => {
+                if (
+                    data?.status === SchedulerJobStatus.COMPLETED ||
+                    data?.status === SchedulerJobStatus.ERROR
+                )
+                    return false;
+                return 2000;
+            },
+            onSuccess: (data) => {
+                if (data?.status === SchedulerJobStatus.ERROR) {
+                    showToastError({
+                        title: 'Could not run SQL query',
+                        subtitle: data.details?.error,
+                    });
+                }
+            },
+            onError: async () => {
+                await queryClient.cancelQueries([
+                    'jobStatus',
+                    sqlQueryJob?.jobId,
+                ]);
+            },
+            enabled: Boolean(sqlQueryJob && sqlQueryJob?.jobId !== undefined),
+        },
+    );
+
+    const { data: sqlQueryResults, isLoading: isResultsLoading } = useQuery<
+        ResultRow[] | undefined,
+        ApiError
+    >(
+        ['sqlQueryResults', sqlQueryJob?.jobId],
+        async () => {
+            const url = scheduledDeliveryJobStatus?.details?.fileUrl;
+            const response = await fetch(url, {
```

Comment:
the `fileUrl` is the whole url, rather than the `fileId`. If I use `lightdashApi` it will prefix the url and I don't want that. 
Would it be ok to use fetch here or should we be better at this?
I can get the `fileId` from the `fileUrl` or I can look at the BE implementation to send back the `fileId` instead of the url.

Reply:
keeping this approach because the URL from `fileUrl` can include a _custom_ s3 endpoint